### PR TITLE
Make image hash available for string substitution

### PR DESCRIPTION
**Pull request by jinfwhuang**
This allows deployment files to get a unique identifier based on the content of the deployment (container image).

### CODE REVIEW

File: skylib/kustomize/kustomize.bzl [hunk 1]
```diff
@@ -234,6 +234,11 @@ def _kustomize_impl(ctx):
                 if "{" in regrepo:
                     regrepo = stamp(ctx, regrepo, tmpfiles, ctx.attr.name + regrepo.replace("/", "_"))
                 template_part += " --variable={}={}@$(cat {})".format(kpi.image_label, regrepo, kpi.digestfile.path)
+
+                # Image hash
+                template_part += " --variable={}=$(cat {} | awk '{{print substr($1,8)}}')".format(str(kpi.image_label)+"__HASH", kpi.digestfile.path)
```
Comment (kzadorozhny) [inline skylib/kustomize/kustomize.bzl:239]: To keep the script more portable, I'd go about without using `awk`.

Here is the same thing done in push rule: https://github.com/adobe/rules_gitops/blob/master/skylib/push.bzl#L97
```
tag = "$(cat {} | cut -d ':' -f 2 | cut -c 1-7)".format(_get_runfile_path(ctx, ctx.outputs.digest))
```

File: examples/helloworld/deployment.yaml [hunk 2]
```diff
@@ -11,6 +11,8 @@ spec:
     metadata:
       labels:
         app: helloworld
+        app_label_unique_hash: "{{//helloworld:image__HASH}}"
```
Comment (kzadorozhny) [inline examples/helloworld/deployment.yaml:14]: How about `.digest` and `.short-digest` suffixes?

**Review by kzadorozhny — CHANGES_REQUESTED** [review]
# Please update `examples/helloworld/k8s_deploy_test` to validate the functionality
# Update `README.md` documentation to reflect the changes in variable substitution behavior

File: skylib/kustomize/kustomize.bzl (hunk 1 above)
Comment (jinfwhuang) [reply skylib/kustomize/kustomize.bzl:239]: Sure. I will update it to use `cut`. I myself was debating which one is more portable `cut` vs. `awk`.

File: examples/helloworld/deployment.yaml (hunk 2 above)
Comment (jinfwhuang) [reply examples/helloworld/deployment.yaml:14]: Sounds goods. I will use digest instead of hash.

**Comment by jinfwhuang** [comment]
1. Updated `examples/helloworld/k8s_deploy_test`.
2. Update README.md

**Review by kzadorozhny — DISMISSED** [review]
Thank you for your contribution.

**Comment by kzadorozhny** [comment]
Formatter check is failing. The rest is LGTM.

File: examples/helloworld/deployment.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ spec:
     metadata:
       labels:
         app: helloworld
+        app_label_image_short_digest: "{{//helloworld:image.short-digest}}"
```
Comment (kzadorozhny) [inline examples/helloworld/deployment.yaml:14]: You can move substitution examples under `annotations`, which don't have value size limitations.

**Review by kzadorozhny — APPROVED** [review]

**Pull request merged**